src/main/java/de/tum/in/www1/artemis/service/connectors/localci/LocalCIBuildJobExecutionService.java

```diff
@@ -228,32 +258,40 @@ private LocalCIBuildResult runScriptAndParseResults(ProgrammingExerciseParticipa
             // empty list for successful tests).
             localCIContainerService.stopContainer(containerName);
             // Delete script file from host system
-            localCIContainerService.deleteScriptFile(participation.getId().toString());
+            localCIContainerService.deleteScriptFile(containerName);
             return constructFailedBuildResult(branch, assignmentRepoCommitHash, testRepoCommitHash, buildCompletedDate);
         }
 
         List<String> testResultsPaths = getTestResultPaths(participation.getProgrammingExercise());
 
         // Get an input stream of the test result files.
         List<TarArchiveInputStream> testResultsTarInputStreams = new ArrayList<>();
+        boolean errorOccurred = false;
         try {
             for (String testResultsPath : testResultsPaths) {
                 testResultsTarInputStreams.add(localCIContainerService.getArchiveFromContainer(containerId, testResultsPath));
             }
         }
         catch (NotFoundException e) {
             // If the test results are not found, this means that something went wrong during the build and testing of the submission.
-            // Stop the container and return a build results that indicates that the build failed.
+            errorOccurred = true;
```

Comment:
Why don't just return here like before?
```suggestion
            return constructFailedBuildResult(branch, assignmentRepoCommitHash, testRepoCommitHash, buildCompletedDate);
```